spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala

```diff
@@ -202,21 +203,12 @@ case class TableDef(
        |  $columnList,
        |  PRIMARY KEY ($primaryKeyClause)
        |)""".stripMargin
-    val ordered = if (clusteringColumns.size > 0)
-      s"$stmt${Properties.lineSeparator}WITH CLUSTERING ORDER BY (${clusteringColumnOrder(clusteringColumns)})"
-    else stmt
-    appendOptions(ordered, options)
-  }
-  private[this] def clusteringColumnOrder(clusteringColumns: Seq[ColumnDef]): String =
-    clusteringColumns.map { col => s"${quote(col.columnName)} ${col.clusteringOrder}"}.mkString(", ")
-
-  def clusterOrder: Seq[ClusteringOrder] = clusteringColumns.map(_.clusteringOrder)
+    val ordered = clusteringColumns.map( col => s"${quote(col.columnName)} ${col.clusteringOrder}")
+      .mkString("CLUSTERING ORDER BY (", ", ",")")
 
-  private[this] def appendOptions(stmt: String, opts: String) =
-    if (stmt.contains("WITH") && opts.startsWith("WITH")) s"$stmt${Properties.lineSeparator}AND ${opts.substring(4)}"
-    else if (!stmt.contains("WITH") && opts.startsWith("AND")) s"WITH ${opts.substring(3)}"
-    else if (opts == "") s"$stmt"
-    else s"$stmt${Properties.lineSeparator}$opts"
+    val orderWithOptions:Seq[String] = if (clusteringColumns.size > 0) options.+:(ordered) else options
```

Comment:
lets just go straight to the treble quote here, I think it may be a bit clearer like
```
val tableOptionsClause = s"WITH $clusteringOrderingClause ${(tableOptions...).mkString(AND)}"
"""$stmt $tableOptionsClause"""